src/storage/src/store_impl.rs

```diff
@@ -117,6 +121,15 @@ impl StateStoreImpl {
                         state_store_stats.clone(),
                     );
                     in_mem_object_store.set_compactor_shutdown_sender(shutdown_sender);
+                } else if let ObjectStoreImpl::Disk(disk_object_store) = object_store.as_ref() {
+                    tracing::info!("start a compactor for local disk object store");
+                    let (_, shutdown_sender) = Compactor::start_compactor(
```

Comment:
Shall we limit the capability of this compactor? e.g., only allow it to compact files on local disk.

Reply:
The compactor is unaware of where the file is coming from. It just fetches table id from meta and data from object store. The code here is only for probably using disk object store as the remote ground true object, similar to using in memory object store.